### PR TITLE
Signup route

### PR DESCRIPTION
**Pull request by MohammadKoolivand**
first part of signup completed. in this part user should enters his number and get code

### CODE REVIEW

File: src/controllers/app/userController.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          return res.status(200).json({
+            message: "successfully saved",
+            id: savedNewUser._id,
+          });
+        });
+      } else {
+        if (findedUser.isActive === true) {
+          return res
+            .status(400)
+            .json({ message: "user is already active in another phone" });
```
Comment (farshid0594) [inline src/controllers/app/userController.js]: change message:
This mobile number has already been registered

File: src/controllers/app/userController.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            message: "successfully updated",
+            id: savedNewUser._id,
+          });
+        });
+      }
+    });
+  } catch (e) {
+    return res.status(500).json({ message: "server error" });
+  }
+};
```
Comment (farshid0594) [inline src/controllers/app/userController.js]: use This code 
const moment = require("moment");

const utils = require("../../helpers/utils");
const User = require("../../models/user");

exports.SignUp = function (req, res) {
  try {
    // get req body
    const { mobile } = req.body;
    // check user exists
    User.findOne({ mobile: mobile }, (err, findedUser) => {
      if (err) {
        return res.status(500).json({ message: "server error" });
      }
      if (findedUser && findedUser.isActive === true) {
        return res
          .status(400)
          .json({ message: "This mobile number has already been registered" });
      }
      let newUser;
      const generatedCode = utils.generateCode();
      const expiredTime = moment().add(2, "minute").format("YYYYMMDDThhmmss");
      if (findedUser === null) {
        //create new User
        newUser = new User({
          mobile: mobile,
          code: {
            text: generatedCode,
            expired: expiredTime,
          },
        });
      } else {
        //updated inActive user
        newUser = findedUser;
        newUser.code = {
          text: generatedCode,
          expired: expiredTime,
        };
      }
      newUser.save((err, savedUser) => {
        if (err) {
          return res.status(500).json({ message: "Server Error" });
        }
        return res.status(200).json({
          message: "successfully updated",
          id: savedUser._id,
        });
      });
    });
  } catch (e) {
    return res.status(500).json({ message: "server error" });
  }
};

**Review by farshid0594 — CHANGES_REQUESTED** [review]
Please resolve reviews

**Review by alikia1995 — APPROVED** [review]

File: src/controllers/app/userController.js (hunk 1 above)
Comment (MohammadKoolivand) [reply src/controllers/app/userController.js]: done

File: src/controllers/app/userController.js (hunk 2 above)
Comment (MohammadKoolivand) [reply src/controllers/app/userController.js]: done

**Review by farshid0594 — APPROVED** [review]

**Pull request merged**